dev_tools/modules.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import dataclasses
+import os
+import sys
+from pathlib import Path
+from typing import List, Dict, Any
+
+_FOLDER = 'folder'
+_PACKAGE_PATH = 'package-path'
+
+
+@dataclasses.dataclass
+class Module:
+    root: Path
+    raw_setup: Dict[str, Any]
+
+    name: str = dataclasses.field(init=False)
+    version: str = dataclasses.field(init=False)
+    top_level_packages: List[str] = dataclasses.field(init=False)
+
+    def __post_init__(self):
+        self.name = self.raw_setup['name']
+        if 'packages' in self.raw_setup:
+            self.top_level_packages = [p for p in self.raw_setup['packages'] if '.' not in p]
+        else:
+            self.top_level_packages = []
+        self.top_level_package_paths = [os.path.join(self.root, p) for p in self.top_level_packages]
```

Comment:
Add `top_level_package_paths` as a dataclass field. Could also add return type annotation `-> None` so that mypy will check this function; I imagine that would've caught that this is assigning to a field that is not defined.

Reply:
Ooops, good catch! Yeah, `-> None` doesn't seem to signal to either mypy or pylint :/